### ISSUE TITLE
Genshin giving the error 31-4302 when loading with fov unlocker

### ISSUE THREAD

**Issue by ZIFIXLeDebutant** [z3lx/genshin-fov-unlock#5]
I had just played this morning before going to class and everything was going perfectly well (I was using 3dmigoto at the same time) then when I got home this evening I noticed that genshin was giving me the 31-4302 error, I tested the game without any mods and no sign of the error, the same with the fps unlocker and after some testing I was able to conclude that the problem was coming from the fov unlocker. Is there any way to fix this? I hate getting motion sickness with the basic fov.

**Comment by z3lx** [comment]
This also happened back in Genshin 4.8 second half of the patch. They removed it in 5.0, but it seems like it's back. To be honest with you, unless hoyo (for some reason) removes it again, finding a workaround seems rather unlikely.

**Comment by ZIFIXLeDebutant** [comment]
I see, it's a real shame, I started playing genshin again just a bit before the 5.0 release after seeing that it was possible to change the fov, but thanks for the clarification tho!

**Comment by z3lx** [comment]
I'll see what can be done about it, but don't expect a quick fix.

**Comment by Mizerka** [comment]
tried some stuff thinking it was just me but yeah same here, only had it for a day lmao

**Comment by z3lx** [comment]
Yeah it actually sucks lol
I'm currently working on a temporary fix that could work, since the integrity check is only performed (afaik) once when you start the game.

**Comment by z3lx** [comment]
I've drafted a [release](https://github.com/z3lx/genshin-fov-unlock/releases/tag/v1.1.3-hotfix) that implements a hotfix for the integrity checks. You do have to jump through a couple extra hoops though lol

**Comment by z3lx** [comment]
Can you provide details? What did you do to set it up by following the readme?

**Comment by z3lx** [comment]
> [...] you now need to manually hook the plugin by pressing <kbd>up arrow</kbd> **when in the loading screen of a domain**.

You're not supposed to hook to the game before getting past the door at the main menu. Log into the game, teleport to a domain, start it, and then, in the domain loading screen, press <kbd>up arrow</kbd>.

**Comment by z3lx** [comment]
The error is not supposed to pop up if the only DLL you're injecting is this plugin. Launch the game via the start game button from `unlockfps_nc.exe`. Do not run this with lanylow's plugin (in case you are).

![unlockfps_nc](https://github.com/user-attachments/assets/b5fc6b0c-4fde-474d-bb58-59f8da991ee4)

**Comment by z3lx** [comment]
Give me a moment. I will record a tutorial to demonstrate.

**Comment by z3lx** [comment]
https://github.com/user-attachments/assets/9abaca8f-c3e9-460d-a5f7-cb907ff2a1e9

**Comment by Mizerka** [comment]
Worked fine for me yesterday btw, thanks.

**Comment by elbadcode** [comment]
hmm injecting your plugin with my own dll injector works perfectly fine for FOV without having to press anything. whats really odd is my code for injecting DLLs is nearly identical to the genshin FPS unlocker so I'm confused.

Actually I just tried using the fps unlocker to launch which does still work for FPS, and it didnt autohook but I didnt need to go to a loading screen to hook and no crashes. Odd

Regardless looks like I need to update my loader to inject shell code or just use the FPS unlocker stub module. I had been using genshin utility for fps and fov

**Comment by Mizerka** [comment]
also im not very smart but the check happens fairly early on into door loading, if I do it around halfway through (after integrity checks) it injects it fine without throwing errors. saves you going into domain, also I'm doing gimi and that didnt break with update, not sure if they inject it differently though.

**Comment by viitana** [comment]
Looks like exiting from co-op sessions back to single player will also trigger the error message and sign the user out.

**Comment by Mizerka** [comment]
yeah co-op event going on atm or serenitea zone will throw error when entering back into single person mode

**Comment by z3lx** [comment]
@elbadcode 
> Actually I just tried using the fps unlocker to launch which does still work for FPS, and it didnt autohook but I didnt need to go to a loading screen to hook and no crashes. Odd

If may or may not crash -- you're essentially just trying to time the hooking process so that it doesn't modify the function while it's being called 2 times a frame, 60+ frames per second. So you were lucky lol

**Comment by z3lx** [comment]
@Mizerka 
> also im not very smart but the check happens fairly early on into door loading, if I do it around halfway through (after integrity checks) it injects it fine without throwing errors. saves you going into domain

Thanks for pointing it out, I'll update the readme!

> also I'm doing gimi and that didnt break with update, not sure if they inject it differently though.

There's only so many ways of injecting dlls -- I'm guessing mhy just isn't checking if d3d11 is modified or not.

**Comment by z3lx** [comment]
@viitana 
> Looks like exiting from co-op sessions back to single player will also trigger the error message and sign the user out.

I'll see what I can do about it. No promises though

**Comment by z3lx** [comment]
Good to hear that it's now working for you. Next time be careful when sharing videos, your UID was exposed.

**Comment by elbadcode** [comment]
> @elbadcode 
> > Actually I just tried using the fps unlocker to launch which does still work for FPS, and it didnt autohook but I didnt need to go to a loading screen to hook and no crashes. Odd
> 
> If may or may not crash -- you're essentially just trying to time the hooking process so that it doesn't modify the function while it's being called 2 times a frame, 60+ frames per second. So you were lucky lol

Hmm either I'm just really lucky or there's another factor because I've launched it like 6 times without a crash

**Comment by z3lx** [comment]
> Hmm either I'm just really lucky or there's another factor because I've launched it like 6 times without a crash

Maybe instead I'm just incredibly unlucky, which would be representative of my gacha luck lol. Originally I just put a startup delay, but it crashed instantly the first time for me. 

> They actually do specifically check d3d11 and dxgi by name. You can use ultimate asi loader as dinput8.dll without issues but if you change it to either of those names it will get removed. Somehow special k wrapper version works as dxgi though. But otherwise 3dmigoto and reshade can be injected just fine so long as they can find the original d3d11 on your PC and are kept out of the game folder

I don't do extensive modding, so I had to look up the repos you were talking about. Honestly, can't really speak much on the subject matter, but it does sound like it has less to do with it being a deliberate check and more of a game engine thing and how it handles graphics APIs internally (?)

> Nah this was to address actual cheats which do hurt them financially.

If they really did want to address actual cheats they'd be doing it through better server-sided heuristic analysis and not this. If a stupid uni student can figure out a workaround in less than half a day's work, either they're doing a comedically bad job or this change is just not intended to be effective against actual cheats.

**Comment by z3lx** [comment]
Integrity check has seemingly been disabled, and I'm guessing it's only a matter of time before mhy re-enables it again. I'll add an autohook option in the config tomorrow.

**Comment by z3lx** [comment]
To avoid getting booted to the login screen, you can unhook the plugin with <kbd>up arrow</kbd>, and then re-hook again it with <kbd>up arrow</kbd> after the "leaving co-op" black screen.

This is not to stop exploiting banners -- it's literally not possible to exploit it since the validation for pulling is all done server-sided.

**Comment by z3lx** [comment]
Fixed for now. If there are bugs or edge cases, you're welcome to report them.
